ZzaDashboard/ViewModel/MainViewModel.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Input;
+using Zza.Data;
+
+namespace ZzaDashboard.ViewModel
+{
+    class MainViewModel : INotifyPropertyChanged
+    {
+        private Customer selectedCustomer;
+        public Customer SelectedCustomer
+        {
+            get => this.selectedCustomer;
+            set
+            {
+                if (value == this.selectedCustomer)
+                    return;
+
+                this.selectedCustomer = value;
+                this.OnPropertyChanged(nameof(this.SelectedCustomer));
+            }
+        }
+
+        private ObservableCollection<Customer> customers;
+        public ObservableCollection<Customer> Customers
+        {
+            get => this.customers;
+            set
+            {
+                if (value == this.customers)
+                    return;
+
+                this.customers = value;
+                this.OnPropertyChanged(nameof(this.Customers));
+            }
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+        protected void OnPropertyChanged(string propertyName)
+        {
+            if (this.PropertyChanged is null)
+                return;
+
+            this.PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+
+        public ICommand SaveCommand { get; set; }
+
+
+        public MainViewModel()
+        {
+            if (DesignerProperties.GetIsInDesignMode(new DependencyObject()))
+                return;
+
+            this.Customers = GetAllCustomers();
+
+            this.SaveCommand = new RelayCommand(SaveCommandExecuted, SaveCommandCanExecute);
+
+        }
+
+        private ObservableCollection<Customer> GetAllCustomers()
+        {
+            string hugeText = string.Empty;
+            ObservableCollection<Customer> customers = new ObservableCollection<Customer>();
+            using (var reader = new StreamReader($@"{Directory.GetCurrentDirectory()}\ZzaPersons.txt"))
+            {
+                hugeText = reader.ReadToEnd();
+            }
+            foreach (var line in hugeText.Split(Environment.NewLine.ToCharArray()))
+            {
+                if (!string.IsNullOrEmpty(line))
+                {
+                    var customer = new Customer();
+                    customer.FirstName = line.Split('~')[0];
```

Comment:
You are splitting the same string several times in the same way. This could lead to performance issues if you don't pay attention to such cases.
It is better to store in a local variable the result of splitting - array.
And then you can call this array by indexer for each item.

Reply:
now I hope you meant this

                string[] lineArr = line.Split('~');
                var customer = new Customer();
                customer.FirstName = lineArr[0];
                customer.LastName = lineArr[1];
                customer.Phone = lineArr[2];
                customers.Add(customer);

Reply:
Exactly